### PR TITLE
Add setup files, Author and Changelog

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: TG1999 <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
@pombredanne wrt to https://github.com/nexB/scancode.io/issues/35, please provide feedback over this

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+author = nexB. Inc. and others
+author_email = info@aboutcode.org
+description = It is a library to reliably fetch code via HTTP, FTP and version control systems.
+long_description = file:README.rst
+url = https://github.com/nexB/fetchcode
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Intended Audience :: Developers
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3 :: Only
```
Comment (majurg) [inline setup.cfg:16]: I forget, did we start this repository with python3?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+package_dir=
+    =src
+packages=find:
+include_package_data = true
+zip_safe = false
+install_requires =
+	attrs
+    packageurl-python
+    requests
+	six
```
Comment (majurg) [inline setup.cfg:31]: Could you make these all 4-space indents?

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,41 @@
+[wheel]
+universal=1
+
+[metadata]
+license_file = apache-2.0.LICENSE
+name = Fetchcode
+author = nexB. Inc. and others
+author_email = info@aboutcode.org
+description = It is a library to reliably fetch code via HTTP, FTP and version control systems.
```
Comment (majurg) [inline setup.cfg:9]: instead of `It`, I would put `fetchcode`

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,41 @@
+[wheel]
+universal=1
+
+[metadata]
+license_file = apache-2.0.LICENSE
+name = Fetchcode
```
Comment (majurg) [inline setup.cfg:6]: We should keep this all lowercase (`fetchcode`). Just to match convention.

**Review by majurg — CHANGES_REQUESTED** [review]
@TG1999 This is looking good, see my few comments.

File: setup.cfg (hunk 1 above)
Comment (TG1999) [reply setup.cfg:16]: Yes we started this repo with python3

File: .travis.yml [hunk 5]
```diff
@@ -1,51 +1,17 @@
...
+  - "3.7"
+  - "3.8"
 
+# Scripts to run at install stage
+install: ./configure
 
-before_cache:
-  - if [ "${TRAVIS_OS_NAME}" = "osx" ]; then brew cleanup; fi
+# Scripts to run at script stage
+script: bin/pytest
```
Comment (majurg) [inline .travis.yml:17]: Change this to `tmp/bin/pytest` and the CI should work :)

**Review by majurg — CHANGES_REQUESTED** [review]

**Review by majurg — COMMENTED** [review]
This all looks good to my eye, however there is a Travis build failure here: https://travis-ci.org/github/nexB/fetchcode/builds/740005065

**Comment by TG1999** [comment]
Hi @majurg how can we resolve it? In earlier config all tests were passing, I can't understand why they are failing now, need some help on that

**Comment by majurg** [comment]
To reiterate our conversation on https://gitter.im/aboutcode-org/discuss:

At first glance it looks like some import errors for some of the internal pip modules.

**Comment by sbs2001** [comment]
This is caused because pytest is collecting the files at https://github.com/TG1999/fetchcode/tree/publish/src/fetchcode/vcs/pip for running doctests. That probably casues some collisions with some internal pip modules . 

One way to fix this would be to be explicit about test discovery. Essentially replacing 

https://github.com/TG1999/fetchcode/blob/b95dd97f75c152cc2e9f88d61f16051bc765f3a7/.travis.yml#L17 to either 
`tmp/bin/pytest --ignore=src/fetchcode/vcs` or simply `tmp/bin/pytest tests `

**Comment by TG1999** [comment]
Thanks @sbs2001

**Comment by TG1999** [comment]
Ping @majurg @pombredanne

**Review by majurg — APPROVED** [review]
@TG1999 Looks good! 

And thanks for the help @sbs2001

**Pull request merged**